src/vm/securitydescriptor.cpp

```diff
@@ -476,4 +476,20 @@ BOOL PEFileSecurityDescriptor::AllowBindingRedirects()
 
 #endif // FEATURE_CORECLR
 
+#if defined(FEATURE_COLLECTIBLE_ALC) && !defined(CROSSGEN_COMPILE)
+SecurityDescriptor::~SecurityDescriptor()
+{
+    // When the AppDomain's LoaderAllocator is not collectible, its LargeHeapHandleTable will
+    // be used for the permission handles
+    if (!m_pLoaderAllocator->IsCollectible())
+    {
+        if (m_hGrantedPermissionSet != NULL)
+            m_pAppDomain->ReleaseObjRefPtrsInLargeTable((OBJECTREF*)((UINT_PTR)m_hGrantedPermissionSet - 1), 1);
```

Comment:
So I should be moving some of the `LoaderAllocator` initialization I added to `Assembly::Create()` into `AppDomain::LoadDomainAssemblyInternal()`? This would allow me to use the correct `LoaderAllocator` when creating the `DomainAssembly` and should fix other cases.